components/patient-data/indexing/src/main/java/org/phenotips/data/indexing/internal/SolrPatientIndexer.java

```diff
@@ -138,4 +169,18 @@ public void reindex()
             this.logger.warn("Failed to search patients for reindexing: {}", ex.getMessage());
         }
     }
+
+    private void addGenesWithStatusToField(SolrInputDocument input, PatientData<Map<String, String>> allGenes,
+        String status, String field)
+    {
+        if (allGenes != null && allGenes.isIndexed()) {
```

Comment:
Patient data comes in 3 flavours (no ice cream flavour :( ). The underlying value can be stored as a single variable value, as a map or as a list. The checks ```isIndexed``` and ```isDictionary``` are there to determine which flavour it is.